src/main/java/com/ericsson/ei/controller/SubscriptionController.java

```diff
@@ -51,21 +54,30 @@
         List<com.ericsson.ei.controller.model.Subscription> subscription);
 
     /**
-     * This method returns the subscriptions for the given subscription names.
+     * This method removes subscriptions from the database. It requires a list of names by using the 'names' parameter
      * 
      */
-    @RequestMapping(value = "/{subscriptionNames}", method = RequestMethod.GET)
-    public ResponseEntity<?> getSubscriptionByNames(
+    @RequestMapping(value = "", method = RequestMethod.DELETE)
+    public ResponseEntity<?> deleteSubscriptions(
+        @RequestParam
+        String subscriptionName);
+
+    /**
+     * This method returns the subscription for the given subscription name.
+     * 
+     */
+    @RequestMapping(value = "/{subscriptionName}", method = RequestMethod.GET)
+    public ResponseEntity<?> getSubscriptionByName(
         @PathVariable
-        String subscriptionNames);
+        String subscriptionName);
 
     /**
-     * This method removes the subscriptions from the database for the given subscription names.
+     * This method removes the subscription from the database for the given subscription name.
      * 
      */
-    @RequestMapping(value = "/{subscriptionNames}", method = RequestMethod.DELETE)
-    public ResponseEntity<?> deleteSubscriptionByNames(
+    @RequestMapping(value = "/{subscriptionName}", method = RequestMethod.DELETE)
+    public ResponseEntity<?> deleteSubscriptionByName(
         @PathVariable
-        String subscriptionNames);
+        String subscriptionName);
```

Comment:
See my previous comment, I think it applies to this endpoint also.

Reply:
So after talking more about this the parameter is now called subscriptionNames in plural.